feat: File I/O instrumentation (#4575)
getsentry/sentry-docs: src/platforms/android/common/performance/instrumentation/automatic-instrumentation.mdx

It's recommended to use `androidx.room` version `2.0.0` or above, since it supports incremental builds.

</Note>

### File I/O Instrumentation

> The [Sentry Android Gradle Plugin](/platforms/android/gradle/) does the tracing auto instrumentation using bytecode manipulation for `java.io.FileInputStream`, `java.io.FileOutputStream`, `java.io.FileReader` and `java.io.FileWriter`.

Reviewer: ```suggestion
The [Sentry Android Gradle Plugin](/platforms/android/gradle/) does tracing auto instrumentation using bytecode manipulation for `java.io.FileInputStream`, `java.io.FileOutputStream`, `java.io.FileReader`, and `java.io.FileWriter`.
```

The plugin replaces the aforementioned classes with custom Sentry-specific implementations.

> The Sentry-specific File I/O implementation starts a span out of the active span, bound to the scope for each File I/O operation. The SDK sets the span `operation` to `file.write`/`file.read` and `description` to the SQL Query, if available.

Author: ```suggestion
The Sentry-specific File I/O implementation starts a span out of the active span, bound to the scope for each File I/O operation. The SDK sets the span `operation` to `file.write`/`file.read` and `description` to `filename (pretty-printed file size)`, e.g. `file.txt (123 kB)`
```

> The Sentry-specific File I/O implementation starts a span out of the active span bound to the scope for each File I/O operation. The SDK sets the span `operation` to `file.write`/`file.read` and `description` to the SQL Query if available.

Reviewer: ```suggestion
The Sentry-specific File I/O implementation starts a span out of the active span, bound to the scope for each File I/O operation. The SDK sets the span `operation` to `file.write`/`file.read` and `description` to the SQL Query, if available.
```
